src/api/fungible_asset.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { GetFungibleAssetMetadataResponse, HexInput, PaginationArgs } from "../types";
+import { AptosConfig } from "./aptos_config";
+import { getFungibleAssetMetadata } from "../internal/fungible_asset";
+
+/**
+ * A class to query all `FungibleAsset` related queries on Aptos.
+ */
+export class FungibleAsset {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Queries the fungible asset metadata
+   *
+   * This query returns the fungible asset metadata for all fungible assets.
+   * It can be filtered by creator address and asset type.
+   *
+   * @param options.pagination Where offset and limit can be specified
+   * @param options.whereCondition Where condition to filter the results
+   * @returns getFungibleAssetMetadata A list of fungible asset metadata
+   */
+  async getFungibleAssetMetadata(args: {
+    options?: {
+      pagination?: PaginationArgs;
+      whereCondition?: {
+        creatorAddress?: HexInput;
+        assetType?: string;
```

Comment:
Can we use the structured type (0x{}::{}::{}) I think it's somewhere else in the code